src/components/ExportToCognitoS3Dialog/index.js

```diff
@@ -0,0 +1,213 @@
+import React, { useEffect, useState } from "react"
+import SimpleDialog from "../SimpleDialog"
+import DataTable from "react-data-table-component"
+import useActiveDatasetManager from "../../hooks/use-active-dataset-manager"
+import isEmpty from "lodash/isEmpty"
+import datasetManagerCognito from "udt-dataset-managers/dist/CognitoDatasetManager"
+import useAuth from "../../utils/auth-handlers/use-auth"
+import { TextField } from "@material-ui/core"
+const redText = { color: "orange" }
+const tableStyle = {
+  marginLeft: "auto",
+  marginRight: "auto",
+  width: "100%",
+}
+
+const expandedAnnotationsColumns = [
+  { name: "Annotations", selector: "annotation" },
+  { name: "Last Modified", selector: "lastModified", sortable: true },
+]
+
+const columns = [{ name: "Projects", selector: "folder", sortable: true }]
+
+const customStyles = {
+  headCells: {
+    style: {
+      paddingLeft: "10px",
+      "font-weight": "bold",
+    },
+  },
+  cells: {
+    style: {
+      paddingLeft: "25px",
+    },
+  },
+}
+
+const ExpandedRow = ({ data }) => {
+  const { rowAnnotations } = data
+  return (
+    <>
+      <DataTable
+        style={{
+          boxSizing: "border-box",
+          paddingLeft: "50px",
+          paddingRight: "10px",
+        }}
+        dense
+        striped
+        noHeader
+        columns={expandedAnnotationsColumns}
+        data={rowAnnotations}
+        noDataComponent='Make sure the project has "samples" folder'
+        pagination={rowAnnotations.length > 10}
+        paginationPerPage={10}
+        paginationRowsPerPageOptions={[10, 20, 25, 50, 100, 200]}
+        customStyles={customStyles}
+      />
+    </>
+  )
+}
+
+export default ({ open, onClose }) => {
+  const [dm, setDm] = useState()
+  const { authConfig } = useAuth()
+  const [projects, setProjects] = useState()
+  const [activeDatasetManager] = useActiveDatasetManager()
+  const [nameProjectToCreate, setNameProjectToCreate] = useState("")
+  const [nameProjectExist, setNameProjectExist] = useState(false)
+  const getCurrentDataset = async () => {
```

Comment:
a function that starts with "get", like "getSomething" has implied immutability, meaning it should not have any side effects. The side effect in this case is calling setNameProjectToCreate. Rename this function to "loadProjectNameFromCurrentDataset"